kanalony-storage-logic/src/main/scala/kanalony/storage/logic/QueryBase.scala

```diff
@@ -13,23 +13,66 @@ abstract class QueryBase[TReq, TQueryRow] extends IQuery {
 
   val dbApi = DbClientFactory
 
-  protected def extractParams(queryParams : QueryParams) : TReq
+  val metricValueHeaderName = "count"
+
+  private[logic] def extractParams(queryParams : QueryParams) : TReq
 
   private[logic] def executeQuery(params : TReq) : Future[List[TQueryRow]]
 
-  protected def getResultHeaders() : List[String]
+  private[logic] def getResultHeaders() : List[String]
```

Comment:
We should indeed incorporate the metric name in the result headers, however since this logic is common to all queries it shouldn't be added to this method (which is implemented implemented by each query). I'll add it to the queryResult processing, since that's the place we determine which columns should make their way to the query result (which will be even more relevant when we have several metrics stored in a single table and we may only wish to display some of them)